source/extensions/transport_sockets/tls/ssl_socket.cc

```diff
@@ -377,6 +377,40 @@ absl::optional<SystemTime> SslSocket::expirationPeerCertificate() const {
   return Utility::getExpirationTime(*cert);
 }
 
+bool SslSocket::peerCertificateChainPresented() const {
+  bssl::UniquePtr<STACK_OF(X509)> cert_chain(SSL_get_peer_cert_chain(ssl_.get()));
```

Comment:
I don't think you need UniquePtr if you don't `X509_chain_up_ref`? Anyway, will see how ASAN says.

Reply:
Addressed in d31dc1e.